feat: add similarity search with distance score

juanfe88 opened the pull request:
Thank you for opening a Pull Request! Before submitting your PR, there are a few things you can do to make sure it goes smoothly:
- [x] Make sure to open an issue as a [bug/issue](https://github.com/googleapis/langchain-google-firestore-python/issues/new/choose) before writing your code!  That way we can discuss the change, evaluate designs, and agree on the general idea
- [x] Ensure the tests and linter pass
- [x] Code coverage does not decrease (if any source code was changed)
- [x] Appropriate docs were updated (if necessary)

Fixes #<issue_number_goes_here> 🦕
I was not the one submitting the issue however I ran into the same problem and figured out I could try and fix it
https://github.com/googleapis/langchain-google-firestore-python/issues/87

juanfe88 committed:
feat: add similarity search with distance score

pl04351820 commented on src/langchain_google_firestore/vectorstores.py:
```diff
@@ -240,6 +240,7 @@ def _similarity_search(
         query: List[float],
         k: int = DEFAULT_TOP_K,
         filters: Optional[BaseFilter] = None,
+        with_scores: Optional[bool] = False,
```
Thanks for adding the support!

Could we pass `distance_result_field` as parameter so that we could calculate the score based on any field (not limit to `distance`)?

juanfe88 replied:
all good I left it as optional with a default value, but it can be overriden. I also added a test for these cases WDYT ?

pl04351820 replied:
Thanks! In this case, do we still need with_scores? Since `distance_result_field` is optional,  `with_scores` field can be implicitly done by distance_result_field. 

(ex: if distance_result_field is specified, return scores)

juanfe88 replied:
You're right. I just refactored it so that we can get the scores just by passing the name of field

juanfe88 committed:
Feat: Allow custom distance field in similarity search results

This change introduces the `distance_result_field` parameter to `similarity_search_with_score` in `FirestoreVectorStore`.

Users can now specify a custom field name for the distance score returned in the results, defaulting to "distance". This provides flexibility and helps avoid potential field name conflicts when results are processed.

The internal `_similarity_search` method was updated to accept and use this custom field name. A new test case verifies this functionality.

juanfe88 committed:
Refactor: Simplify score handling in _similarity_search

Remove the `with_scores` parameter from the `_similarity_search`
method. The `distance_result_field` parameter now solely determines
if distance scores should be calculated and returned by the underlying
`find_nearest` call.

This change simplifies the method's API and internal logic by making
the provision of `distance_result_field` the explicit way to request
scores. The `similarity_search_with_score` method has been updated
accordingly.

juanfe88 committed:
Style: Fix Linting on updated files

pcostell committed:
Merge branch 'main' into feature/add_similarity_with_score

juanfe88 committed:
Merge branch 'main' into feature/add_similarity_with_score
